packages/apps/patterns/react-appkit/src/components/StatusIndicator/StatusIndicator.tsx

```diff
@@ -4,26 +4,45 @@
 
 import React from 'react';
 
-import { Tooltip, useTranslation, valenceColorText, mx } from '@dxos/react-uikit';
+import { useClient, useNetworkMode } from '@dxos/react-client';
+import { Tooltip, useTranslation, valenceColorText, mx, Button } from '@dxos/react-uikit';
 
 // TODO(burdon): Extend to show heartbeat, network status, etc.
 // TODO(burdon): Merge with ErrorBoundary indicator since overlaps.
 export const StatusIndicator = ({ status }: { status: boolean }) => {
   const { t } = useTranslation();
+  const client = useClient();
+  const networkMode = useNetworkMode();
   return (
-    <div role='none' className={mx('fixed bottom-4 right-4', valenceColorText('success'))}>
-      {status && (
-        <span className='flex h-3 w-3'>
-          <span className='animate-ping absolute inline-flex h-full w-full rounded-full bg-green-600 opacity-75'></span>
-          <span className='relative inline-flex rounded-full h-3 w-3 bg-green-700'></span>
-        </span>
-      )}
-      {!status && (
-        <Tooltip content={t('heartbeat stalled')}>
+    <div role='none' className={mx('fixed bottom-4 right-4')}>
+      <div role='none' className={mx('fixed bottom-4 right-4', valenceColorText('success'))}>
+        {status && (
           <span className='flex h-3 w-3'>
-            <span className='relative inline-flex rounded-full h-3 w-3 bg-red-500'></span>
+            <span className='animate-ping absolute inline-flex h-full w-full rounded-full bg-green-600 opacity-75'></span>
+            <span className='relative inline-flex rounded-full h-3 w-3 bg-green-700'></span>
           </span>
-        </Tooltip>
+        )}
+        {!status && (
+          <Tooltip content={t('heartbeat stalled')}>
+            <span className='flex h-3 w-3'>
+              <span className='relative inline-flex rounded-full h-3 w-3 bg-red-500'></span>
+            </span>
+          </Tooltip>
+        )}
+      </div>
+      {networkMode && (
+        <>
+          {' '}
+          <span className='ml-2'>{networkMode.mode === 0 ? 'OFFLINE' : 'ONLINE'}</span>
+          <Button
+            onClick={async () => {
+              const newMode = networkMode.mode === 0 ? 1 : 0;
```

Comment:
whats up with enums and react?

Reply:
<img width="756" alt="image" src="https://user-images.githubusercontent.com/36420699/209722807-e13faa6d-74b4-4d99-962d-218b5a18fc6e.png">

Reply:
That is an error from @dxos/kai

`@dxos/protocols` is included in `dependencies` in package.json
`@dxos/protocols` is also included in `optimizeDeps` in vite.config.ts